presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -444,6 +445,11 @@ public HiveSessionProperties(HiveClientConfig hiveClientConfig, OrcFileWriterCon
                         FAIL_FAST_ON_INSERT_INTO_IMMUTABLE_PARTITIONS_ENABLED,
                         "Fail fast when trying to insert into an immutable partition. Increases load on the metastore",
                         hiveClientConfig.isFailFastOnInsertIntoImmutablePartitionsEnabled(),
+                        false),
+                booleanProperty(
+                        USE_LIST_DIRECTORY_CACHE,
+                        "Use list directory cache if available when set to true",
+                        true,
```

Comment:
Do we wanna set the default value based on hiveClientConfig.getFileStatusCacheExpireAfterWrite/getFileStatusCacheMaxSize/getFileStatusCacheTables?

Reply:
Don't we want it to be backed by a config? Not sure what James is saying.

Reply:
This sounds reasonable, I'll add that.

@jainxrohit : I think @highker is suggesting adding default value backed by a config, which is currently lacking :D

Reply:
Changed default value to `!hiveClientConfig.getFileStatusCacheTables().isEmpty()`